### ISSUE TITLE
Flutter stops animations even when screen is only partially obscured

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#52486]
See: b/151332834

When showing native bottom sheet or alert dialogs, screen is only partially obscured. If there's an ongoing animation, it stops when the native UI starts displaying leading to jarring effects. See repro:

```dart
import 'package:flutter/material.dart';
import 'package:url_launcher/url_launcher.dart';

void main() => runApp(
    MaterialApp(
      home: MyApp(),
      theme: ThemeData(
        splashFactory: InkRipple.splashFactory,
      ),
    )
);

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  _launchURL() async {
    const url = 'tel://9000000';
    if (await canLaunch(url)) {
      // This uses url launcher plugin to display an overlay.
      await launch(url);
    } else {
      throw 'Could not launch $url';
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Flutter"),
      ),
      body: Center(
          child: FlatButton(
            child: Text("A long launch URL"),
            shape: RoundedRectangleBorder(
              side: BorderSide(
                color: Colors.blue,
                width: 1,
              ),
              borderRadius: BorderRadius.all(Radius.circular(24)),
            ),
            onPressed: _launchURL,
          )
      ),
    );
  }
}
```

Video is attached. Ripple gets stuck in the middle when the overlay is shown.

[RPReplay_Final1583912940.mp4.gz](https://github.com/flutter/flutter/files/4325907/RPReplay_Final1583912940.mp4.gz)

**Comment by jonahwilliams** [comment]
It looks like opening this bottom sheet is putting flutter into the paused or detached lifecycle state (https://github.com/flutter/engine/blob/d0c2418dfed3e828c1a003ff8ac4a1baaf5e7ae8/lib/ui/window.dart#L173) instead of "inactive".

The scheduler which controls frame will stop pumping them if  we're in these states: 
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/scheduler/binding.dart#L336

We would need to verify which lifecycle state they're in when the native UIView is shown

**Comment by jonahwilliams** [comment]
We have a lifecycle test here: https://github.com/flutter/engine/blob/1f1e2ba58e6f62ab9c3808fada9769a3f23a3a77/testing/scenario_app/ios/Scenarios/ScenariosTests/AppLifecycleTests.m

But its marked as skipped.

cc @xster @dnfield

**Comment by mehmetf** [comment]
Yes. The tricky part is we would likely want to be in detached state if the entire screen is obscured. Hopefully there's a way to distinguish that in all platforms we support.

**Comment by chunhtai** [comment]
The detached state is when flutter app is running without a view. In this example, the view is still there. I checked the lifecycle state sequence, it goes from resume -> inactive -> pause. We disabled frame when it become pause state. Ideally, the app should stay in inactive when the view is obscured but still visible.

**Comment by xster** [comment]
I only have an iPad at home but in my case, it's only going from resume <-> inactive and I'm still seeing the same issue. Looks like we're having a mapping issue (though either way, unrelated to detached, which is the engine not having a VC attached, not this case). 

Somewhat related to https://github.com/flutter/flutter/issues/45131 also.

We've been so historically defensive about stopping things because we're worried about triggering a background opengl crash. 

I didn't spend a lot of time testing yet but inactive seems appropriate for this case and pseudo obstructions like control center etc. 

Wonder what we're doing differently though @chunhtai

**Comment by dnfield** [comment]
Seems related to https://github.com/flutter/flutter/issues/45595

**Comment by nt4f04uNd** [comment]
Confirm this happens to my Flutter application on iOS 14 (haven't tested) other versions, when I open the control center
